src/block-decorations-presenter.js

```diff
@@ -0,0 +1,150 @@
+/** @babel */
+
+const {CompositeDisposable, Emitter} = require('event-kit')
+const LineTopIndex = require('./linear-line-top-index')
+
+module.exports =
+class BlockDecorationsPresenter {
```

Comment:
I opted to move block decorations handling out of `TextEditorPresenter` because it seemed like this code felt quite tight and cohesive. I am a bit unsure about the naming, though, because this doesn't seem to prepare any state for the view but rather to encapsulate information about block decorations + measurements and provide a cleaner API on top of them. Maybe this is just a `BlockDecorationsManager`? (although it feels a bit generic)